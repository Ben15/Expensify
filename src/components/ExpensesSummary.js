import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import { getExpensesTotal } from '../selectors/expenses-total'
import numeral from 'numeral'


export const ExpensesSummary = (props) => {

    return (
        <div>
            <p>{`You have ${props.expenses.length} expenses of a total of ${numeral(getExpensesTotal(props.expenses) / 100).format('$0,0.00')}` }</p>
        </div>
    )
}

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
}

export default connect(mapStateToProps)(ExpensesSummary)