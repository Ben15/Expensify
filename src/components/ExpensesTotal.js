
export const getExpensesTotal = (expenses) => {
    let total = 0
    if(expenses){
        expenses.map((expense) => {
            total += expense.amount
        })
        
        return total

    } else {
        return total
    }
}

