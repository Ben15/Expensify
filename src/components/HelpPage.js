import React from 'react';

export const HelpPage =() => (
    <div>
    Want some help, no problemo.
    </div>
)