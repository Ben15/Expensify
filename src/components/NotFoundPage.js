import React from 'react';
import {Link, NavLink} from 'react-router-dom'

export const NotFoundPage =() => (
    <div>
    404! - <Link to="/">Go Home</Link>
    </div>
)