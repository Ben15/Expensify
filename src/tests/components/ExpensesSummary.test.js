import React from 'react'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import expenses from '../fixtures/expenses'


test('should render ExpensesSummary component with multiple expenses correctly',() => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses}/>)
    expect(toJSON(wrapper)).toMatchSnapshot()
})

test('should render ExpensesSummary component with a single expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]}/>)
    expect(toJSON(wrapper)).toMatchSnapshot()
})