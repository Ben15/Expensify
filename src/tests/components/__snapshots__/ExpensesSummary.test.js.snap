// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary component with a single expense correctly 1`] = `
<div>
  <p>
    You have 1 expenses of a total of $1.95
  </p>
</div>
`;

exports[`should render ExpensesSummary component with multiple expenses correctly 1`] = `
<div>
  <p>
    You have 3 expenses of a total of $10,951.95
  </p>
</div>
`;
